flow/project.py

```diff
@@ -263,10 +273,13 @@ class _JobOperation(object):
         :class:`dict`
     """
 
-    def __init__(self, id, name, job, cmd, directives=None):
+    def __init__(self, id, name, jobs, cmd, directives=None):
         self._id = id
         self.name = name
-        self.job = job
+
+        assert isinstance(jobs, list)
+
+        self._jobs = jobs
```

Comment:
A user will be able to return aggregates or yield them in any type. We internally convert those generators/Iterables to "nested list"
This was discussed earlier too but for now, I think consistency should be the key. For example, internally, if we use lists somewhere then we'll be using list throughout the code.
If @b-butler agrees, I'll be happy to change lists to tuples.

Reply:
If you want an input to be a `list`, it's better to _convert to a list_ than to just _assert that it is a list_. That's why I think `self._jobs = list(jobs)` (conversion, not assertion) is the right approach.

Changing to `tuple` is a separate question. I think it would be the better choice from a data structure perspective -- the aggregate's jobs should be immutable, and as a secondary consideration, `tuple` is almost always faster than `list`.

Reply:
I agree with @bdice that a `tuple` is likely the preferred data structure, but I don't agree that *converting is better than asserting* since the former triggers a potential copy operation. Since this is an internal class we can make certain assumptions about the passed arguments and the way to explicitly declare those assumptions would be to use an `assert` as was done here.

Reply:
If we want to use tuples for immutability which makes sense here we would want the structure of aggregation to be a generator or list of aggregate tuples. I don't have a strong preference for the assert. Ideally we wouldn't need an assert or explicit conversion.